tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -229,9 +239,121 @@ setup() {
     assert_rootfs_count "$node" "$sandbox_id" "1"
 }
 
+function setup_kbs_credentials() {
+    image=$1
+    user=$2
+    password=$3
+
+    if [ "${KBS}" = "false" ]; then
+        skip "Test skipped as KBS not setup"
+    fi
+
+    if [[ -z ${user} || -z ${password} ]]; then
+        # TODO - is ${GITHUB_ACTION} enough to know we are running in CI?
+        if -n ${GITHUB_ACTION}; then
+            die "User and/or password not supplied to authenticated registry test"
+        else
+            skip "running local and missing user/password"
+        fi
+    fi
+
+    REGISTRY_CREDENTIAL_ENCODED=$(echo "${user}:${password}" | base64 -w 0)
+
+    auth_json=$(echo "{
+        "auths": {
+            "${image}": {
+                "auth": "${REGISTRY_CREDENTIAL_ENCODED}"
+            }
+        }
+    }" | base64 -w 0)
+
+    kbs_set_allow_all_resources
+
+    kbs_set_resource "default" "credentials" "test" "${auth_json}"
+}
+
+function create_pod_with_private_image() {
+    image=$1
+    auth_path_set=${2:-true}
+
+    local CC_KBS_ADDR
+    export CC_KBS_ADDR=$(kbs_k8s_svc_http_addr)
+    kernel_params_annotation="io.katacontainers.config.hypervisor.kernel_params"
+    kernel_params_value="agent.guest_components_rest_api=resource"
+
+    if [ $auth_path_set == true ]; then
+        kernel_params_value+=" agent.aa_kbc_params=cc_kbc::${CC_KBS_ADDR}"
+        kernel_params_value+=" agent.image_registry_auth=kbs://default/credential/test"
```

Comment:
Hi @stevenhorsman !

It should be `default/credentials/test` (`credentials` plural) to match https://github.com/kata-containers/kata-containers/pull/9904/commits/298646d8f124078b7cc675e31631ea354c5937dd#diff-ec67ad40f7cab8f9dec0df4ac19ce973352a6a49d63ec5b77e3a18a0bf93acebR272

Reply:
Great spot!